API/controllers/authControllers.go

```diff
@@ -59,15 +76,24 @@ var CreateAccount = func(w http.ResponseWriter, r *http.Request) {
 		account := &models.Account{}
 		err := json.NewDecoder(r.Body).Decode(account)
 		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
 			u.Respond(w, u.Message(false, "Invalid request"))
 			return
 		}
-		resp, e := account.Create()
+
+		callerUser := getUserFromToken(w, r)
+		if callerUser == nil {
+			return
+		}
+
+		resp, e := account.Create(callerUser.Roles)
 		switch e {
```

Comment:
I think there is no point in returning string status that are then converted into http status code, we might as well return the status code directly. This was already here before the PR so I will still approve but we should consider changing this later.